File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    total = brown + yellow
    
    for column in range(2, total + 1):
        if (total / column) % 1 == 0:
            row = total // column
            if row >= column and (2 * row) + (2 * column) == brown + 4:
                return [row, column]

File: 프로그래머스/lv2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let height = 1; // 높이 1부터 시작
    let width = yellow / height; // 높이에 따른 yellow 넓이 설정
    while(width >= height) { // 높이가 넓이를 넘어가지 않을 때
        if(brown === (width + 2) * 2 + height * 2){
            return [width + 2, height + 2];
        } else {
            height += 1; // 높이 + 1
            width = yellow / height; // 높이 변화에 따른 넓이 수정
        }
    }
}